### PR TITLE
Default values for python 3

### LINKED ISSUE #77: Filename generation reset
Output file names generation is driving me crazy. When I try to render two documents in the same python script then it seems available filenames are not reset, although I recreate *all* objects. Say I have inside ``/tmp/test.py``:
```python
import os
from plasTeX.TeX import TeX, TeXDocument
from plasTeX.Config import config
from plasTeX.Renderers.XHTML import Renderer

cur_dir = os.getcwd()
try:
    os.makedirs('build')
except:
    pass

doc = TeXDocument(config=config)
tex = TeX(doc)
tex.input("\\documentclass{article}\\begin{document}Hello.\\end{document}")
doc = tex.parse()
doc.userdata['working-dir'] = '.'

renderer = Renderer()
os.chdir('build')
renderer.render(doc)

os.chdir(cur_dir)
try:
    os.makedirs('build2')
except:
    pass
doc = TeXDocument(config=config)
tex = TeX(doc)
tex.input("\\documentclass{article}\\begin{document}Hello.\\end{document}")
doc = tex.parse()
doc.userdata['working-dir'] = '.'

renderer = Renderer()
os.chdir('build2')
renderer.render(doc)
```
The first rendering renders to ``build/index.html`` but the second one renders to ``build2/sect0001.html``!!!

I don't understand what I'm doing wrong but it is a huge problem when writing unittests. I need my tests to be isolated from each other but, whatever I do, some devil remembers I already rendered some ``index.html``. I tried using ``deepcopy`` everywhere, using ``del`` after using every objects but nothing worked.

### PR DESCRIPTION
**Pull request by PatrickMassot**
This is the python3 twin of #78 

Fixes #77. Remove mutable in function default values except in few cases where the
value is copied anyway. This is a twin of 28058f0d from the python2
branch

### CODE REVIEW

File: plasTeX/ConfigManager/__init__.py [hunk 1]
```diff
@@ -316,6 +316,7 @@ def get(self, option, raw=0, vars={}):
         value of the option
 
         """
+        vars = vars or None
```
Comment (kesmit13) [inline plasTeX/ConfigManager/__init__.py]: I may have missed this in the Python 2 review, should this be 'vars or {}'?

File: plasTeX/ConfigManager/__init__.py [hunk 2]
```diff
@@ -685,9 +688,9 @@ def readfp(self, fp, filename=None):
         self.__read(fp, filename)
         return self
 
-    def get(self, section, option, raw=0, vars={}):
+    def get(self, section, option, raw=0, vars=None):
         """ Get an option value for a given section """
-        return self[section].get(option, raw, vars)
+        return self[section].get(option, raw, vars or None)
```
Comment (kesmit13) [inline plasTeX/ConfigManager/__init__.py]: vars or {}?

File: plasTeX/TeX.py [hunk 3]
```diff
@@ -186,6 +186,7 @@ def loadPackage(self, myfile, options={}):
         options -- options passed to the macro which is loading the package
 
         """
+        options = options or None
```
Comment (kesmit13) [inline plasTeX/TeX.py]: options or {}?

**Comment by PatrickMassot** [comment]
I'm sorry I was tired.

**Pull request merged**